RHOSP-18.0.3/cinder/powerflex/README.md

```diff
@@ -0,0 +1,295 @@
+stringData:
+  powerflex-secrets.conf: |
+    [powerflex]
+    san_ip=PF_Manager_IP
+    san_login=PF_Manager_login
+    san_password=PF_Manager_password
+type: Opaque
+```
+
+* Apply the CR file by
+```yaml
+oc create -f powerflex-secret.yaml
+```
+
+* Verify the secret is created
+```yaml
+oc describe secret/cinder-volume-powerflex-secrets
+```
+For full detailed instruction of all options please refer to [PowerFlex Backend Configuration](https://docs.openstack.org/cinder/wallaby/configuration/block-storage/drivers/dell-emc-powerflex-driver.html).
+
+* Edit the existing OpenStackControlPlane CR
+```
+oc edit openstackcontrolplane
+```
+Modify the cinderVolumes section as follows
+```
+...
+cinderVolumes:
+        powerflex:
+          containerImage: registry.redhat.io/rhosp-dev-preview/openstack-cinder-volume-rhel9:18.0
+          customServiceConfig: |
+            [powerflex]
+            volume_driver = cinder.volume.drivers.dell_emc.powerflex.driver.PowerFlexDriver
+            volume_backend_name = powerflex
+            powerflex_storage_pools = Domain1:Pool1,Domain2:Pool2
+          customServiceConfigSecrets:
+          - cinder-volume-powerflex-secrets
+          debug:
+            service: false
+          networkAttachments:
+          - storage
+          replicas: 1
+          resources: {}
+...
+```
+Save and exit the file, the prompt tells you that the CR was edited. Verify the pod is running
+
+```
+oc get pods -n openstack | grep powerflex
+
+[admin@rhadmin ~]$ oc get pods -n openstack | grep powerflex
+cinder-volume-powerflex-0                                       2/2     Running     0             4d
+[admin@rhadmin ~]$
+
+```
+
+### Prepare custom volume mappings for connector configuration 
+
+PowerFlex SDC needs to be installed on the following nodes:
+* Controller nodes 
+* Compute nodes
+
+To install SDC on the controller nodes, deploy the Dell Container Storage Module operator. From the OpenShift operator hub, search Dell Container Storage modeules and pick the certified operator. Follow the wizard with all the deault values and install the operator. 
+
+Follow the documentation located here https://dell.github.io/csm-docs/docs/deployment/csmoperator/drivers/powerflex/ to install a PowerFlex CSI driver for the remaining steps.
+
+Create a config json file as follows
+
+```ini
+vi config.json
+[
+     {
+         "username": "PF_Manager_login",
+         "password": "PF_Manager_password",
+         "systemID": "SYSTEM_ID",
+         "endpoint": "PF_MGR-IP:443",
+         "insecure": true,
+         "isDefault": true,
+         "mdm": "MDM_IP",
+         "nasName": "none"
+      }
+ ]
+```
+
+Create a secret using
+
+```
+oc create secret generic vxflexos-config -n vxflexos --from-file=config=./config.json
+```
+
+From the sample available at https://github.com/dell/csm-operator/blob/main/samples/storage_csm_powerflex_v2110.yaml , create a CR file. Replace the MDM virtual IP by your environment.
```

Comment:
I guess latest version is 2130 as you can see here: https://github.com/dell/csm-operator/tree/main/samples